OpenSim/Moco/MocoGoal/MocoContactImpulseTrackingGoal.cpp

```diff
@@ -179,76 +175,84 @@ void MocoContactTrackingGoal::initializeOnModelImpl(const Model& model) const {
         groupInfo.refExpressedInFrame = nullptr;
         if (extForce.get_force_expressed_in_body() != "ground") {
             const auto& forceExpressedInBody =
-                    extForce.get_force_expressed_in_body();
+                extForce.get_force_expressed_in_body();
             if (model.hasComponent<PhysicalFrame>(forceExpressedInBody)) {
                 groupInfo.refExpressedInFrame =
-                        &model.getComponent<PhysicalFrame>(
-                                forceExpressedInBody);
-            } else if (model.hasComponent<PhysicalFrame>(
-                               "./bodyset/" + forceExpressedInBody)) {
+                    &model.getComponent<PhysicalFrame>(
+                        forceExpressedInBody);
+            }
+            else if (model.hasComponent<PhysicalFrame>(
+                "./bodyset/" + forceExpressedInBody)) {
                 groupInfo.refExpressedInFrame =
-                        &model.getComponent<PhysicalFrame>(
-                                "./bodyset/" + forceExpressedInBody);
-            } else {
+                    &model.getComponent<PhysicalFrame>(
+                        "./bodyset/" + forceExpressedInBody);
+            }
+            else {
                 OPENSIM_THROW_FRMOBJ(Exception,
-                        "Could not find '{}' in the model or the BodySet.",
-                        forceExpressedInBody);
+                    "Could not find '{}' in the model or the BodySet.",
+                    forceExpressedInBody);
             }
         }
 
         m_groups.push_back(groupInfo);
 
         // Check to see if the model contains a MocoScaleFactor associated with
         // this contact group.
-        std::array<RefPtrMSF, 3> theseScaleFactorRefs;
-        for (int j = 0; j < 3; ++j) {
-            bool foundScaleFactor = false;
-            std::pair<std::string, int> key(group.get_external_force_name(), j);
-            for (const auto& scaleFactor : scaleFactors) {
-                if (m_scaleFactorMap[key] == scaleFactor.getName()) {
-                    theseScaleFactorRefs[j] = &scaleFactor;
-                    foundScaleFactor = true;
-                }
-            }
-            // If we didn't find a MocoScaleFactor for this force direction, set
-            // the reference pointer to null.
-            if (!foundScaleFactor) {
-                theseScaleFactorRefs[j] = nullptr;
+        RefPtrMSF thisScaleFactorRef;
+        bool foundScaleFactor = false;
+        std::string key(group.get_external_force_name());
+        for (const auto& scaleFactor : scaleFactors) {
+            if (m_scaleFactorMap[key] == scaleFactor.getName()) {
```

Comment:
Similar to above, you could just do `m_scaleFactorMap[group.get_external_force_name()]`. In this case it might be cleaner to leave it the way it is, but it does create an extra variable. Up to you.